libnopegl/src/pipeline_compat.c

```diff
@@ -317,6 +318,42 @@ int ngli_pipeline_compat_update_dynamic_offsets(struct pipeline_compat *s, const
     return 0;
 }
 
+void ngli_pipeline_compat_apply_reframing_matrix(struct pipeline_compat *s, int32_t index, const struct image *image, const float *reframing)
+{
+    if (index == -1)
+        return;
+
+    ngli_assert(index >= 0 && index < s->compat_info->nb_texture_infos);
+    const struct pgcraft_texture_info *info = &s->compat_info->texture_infos[index];
+    const struct pgcraft_texture_info_field *fields = info->fields;
+
```

Comment:
You can early exit here if `fields[NGLI_INFO_FIELD_COORDINATE_MATRIX].index == -1`.

Reply:
Added, thanks